src/Development/IDE/Core/Shake.hs

```diff
@@ -97,9 +97,11 @@ data ShakeExtras = ShakeExtras
     ,publishedDiagnostics :: Var (HMap.HashMap NormalizedUri [Diagnostic])
     -- ^ This represents the set of diagnostics that we have published.
     -- Due to debouncing not every change might get published.
-    ,positionMapping :: Var (HMap.HashMap NormalizedUri (Map TextDocumentVersion PositionMapping))
+    ,positionMapping :: Var (HMap.HashMap NormalizedUri (Map TextDocumentVersion (PositionMapping, PositionMapping)))
     -- ^ Map from a text document version to a PositionMapping that describes how to map
     -- positions in a version of that document to positions in the latest version
```

Comment:
Unrelated to this change, but is `positionMapping` ever flushed? It's a space leak that grows with every edit, isn't it?

Reply:
Yes it grows every edit, I have some thoughts about doing some garbage collection but now the growth is only linear it is not as serious a problem. We don't end up with 1gb of residency just from the position map. 

For example, we could remove the whole position map from removing a file from FOI as well as flush out part of the `Values` map.

Reply:
Note that we do have some code for this in DAML https://github.com/digital-asset/daml/blob/master/compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs#L946